From: Miao Xie <miaox@cn.fujitsu.com>
Date: Fri, 21 Dec 2012 10:38:50 +0000
Patch-mainline: 3.8
Git-commit: 97547676570b3bd908560741315bf4b7d635bcf5
References: FATE#312888
Subject: [PATCH] Btrfs: fix missing write access release in
 btrfs_ioctl_resize()

We forget to give up the write access after we find some device operation
is going on. Fix it.

Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1346,6 +1346,7 @@ static noinline int btrfs_ioctl_resize(s
 	if (atomic_xchg(&root->fs_info->mutually_exclusive_operation_running,
 			1)) {
 		pr_info("btrfs: dev add/delete/balance/replace/resize operation in progress\n");
+		mnt_drop_write_file(file);
 		return -EINPROGRESS;
 	}
 
